server/src/main/java/org/apache/kafka/server/config/KRaftConfigs.java

```diff
@@ -125,7 +124,7 @@ public class KRaftConfigs {
     public static final ConfigDef CONFIG_DEF =  new ConfigDef()
             .define(METADATA_SNAPSHOT_MAX_NEW_RECORD_BYTES_CONFIG, LONG, METADATA_SNAPSHOT_MAX_NEW_RECORD_BYTES, atLeast(1), HIGH, METADATA_SNAPSHOT_MAX_NEW_RECORD_BYTES_DOC)
             .define(METADATA_SNAPSHOT_MAX_INTERVAL_MS_CONFIG, LONG, METADATA_SNAPSHOT_MAX_INTERVAL_MS_DEFAULT, atLeast(0), HIGH, METADATA_SNAPSHOT_MAX_INTERVAL_MS_DOC)
-            .define(PROCESS_ROLES_CONFIG, LIST, Collections.emptyList(), ConfigDef.ValidList.in("broker", "controller"), HIGH, PROCESS_ROLES_DOC)
+            .define(PROCESS_ROLES_CONFIG, LIST, List.of("broker"), ConfigDef.ValidList.in("broker", "controller"), HIGH, PROCESS_ROLES_DOC)
```

Comment:
Would it make sense to make the config mandatory instead of having a default?

Reply:
yes, this configuration should be mandatory.

Reply:
+1

Reply:
Let's make sure the error message is also good when the config isn't provided.

Reply:
And we should have this set in the default server.properties we ship with.

Reply:
Thanks for review. I update the validation for `process.roles`. It's required configuration now.